pkg/operator/api/api.go

```diff
@@ -192,7 +200,23 @@ func (a *apiServer) ListComponents(ctx context.Context, in *operatorv1pb.ListCom
 		Components: [][]byte{},
 	}
 	for i := range components.Items {
+		// By default assume that it is not a component for control plane service.
+		controlPlaneComp := false
 		c := components.Items[i] // Make a copy since we will refer to this as a reference in this loop.
+		// If the component is in the same namespace as the operator AND has a scope defined with prefix dapr-, then it is a control plane component.
```

Comment:
Control plane components have identity today so there needn't be any special casing for them. I'd definitely be in favor of adding logic on the server side to filter out Components with a scope that don't match the client, however this must come from the gRPC context identity.

It seems like we are making magic scope values here and as a reviewer it is not clear why we are doing this, and why the identity we have today cannot be used to achieve what we want. I'd like to see a proposal so we can make a decision on a significant API change.

Reply:
My comment isn't about the identity passed to the operator, but rather the value of the scope. To make sure that components scoped to control plane services aren't sent to sidecars, a simple solution is to use the scope `dapr:<foo>` which is invalid for sidecars.

Reply:
What is the reason that we would like to make a distinction between control plane and sidecar clients, rather than using the control plane identity itself, i.e. `dapr-placement`, `dapr-sentry`, etc.

Reply:
Correct me if i'm wrong. scopes are not authenticated. an app (sidecar) can choose app-id "dapr-sentry" and it would then be able to receive components scoped for that.

Reply:
Correct, there is currently _no_ authorization that is happening on the operator side for controlling what component resources are sent to clients. Sidecars are currently being sent all resources in that namespace. The sidecar is currently self electing whether the component should be `init`ed or not. I'm definitely in favor of this authorization happening in the operator, and for all resource types.

Reply:
Thanks for confirming. So, the idea was that using `dapr:<foo>` as scope was a cheap but effective way to obtain some "access control", since no sidecar can have an app-id with a `:`.

Other options considered included:

- Adding a new field to the Component CRD such as "controlPlane" to indicate it's for control plane services only. This however requires changing the CRD which is not ideal
- Ignoring access control, but that may allow sidecars to get access to components they shouldn't see. Assuming the component type isn't supported by the sidecar the risk shouldn't be significant, but it still felt like it'd be nicer to have some separation

Reply:
I think we agree that we should have server side authorization on scoped resources and we should implement this as it should exist today already. If the fundamental intention of this PR is to to do this, then we can merge https://github.com/dapr/dapr/pull/7111, and then follow on to include a function to filter non-matching scopes.

I question why we need any concept of "control plane" service as a scope `group` concept when we have a well defined list of control plane services and their identities. Equally, we need not need to separate or special case control plane and side car identities at all- they are both represented as SPIFFE IDs when connecting to peers over gRPC mTLS.

As a reviewer I cannot understand where the motivation is coming from where we need non-known control plan services which require loading of components- this should be raised as a proposal. Remember that control plane service identities _must_ be known before deployment or must be discoverable, else mTLS  fundamentally itself will not work.